Check `args` slice bounds in `get` and `put` functions

waits opened the pull request:
This adds a bounds check before accessing the `args` slice in the `get` and `put` functions. This has two effects:

1. Displays a friendly error when the first argument (the source path) is not specified, instead of panicking.
2. Allows these commands to work intuitively without a second argument (the destination path) by defaulting the destination path to be the base segment of the source path. This was already partially written into the `get` function but it didn't work and panicked instead.

Fixes #10.

diwakergupta commented on cmd/get.go:
```diff
@@ -27,11 +28,21 @@ import (
 
-	if dst == "" {
+	var dst string
+	if len(args) < 2 {
```
Given the check above, this can really only be `len(args) == 1` right?

waits replied:
Thanks, I didn't catch that.

diwakergupta commented on cmd/get.go:
```diff
@@ -27,11 +28,21 @@ import (
+	var dst string
+	if len(args) < 2 {
+		// If a destination is not specified use the base of the source path
 		dst = path.Base(src)
```
Why not just initialize dst to Base(src) and then use args[1] if len == 2. On that note, should we also error out if len > 2?

waits replied:
You're right, that's much cleaner. I thought assigning `dst` twice would be unnecessary, but it really just cluttered up the code with an extra `else`.

waits replied:
I agree we should error out with too many arguments. Most other CLI tools do the same I believe.

waits commented:
@diwakergupta I added two commits based on your feedback.

- I'm returning an error if 0 or > 2 arguments are passed. Is my one error message good enough for both conditions?
- With those bounds known, I only need to check if `len == 2`.
- I also replaced `len(args) < 1` with `len(args) == 0` because the [spec](https://golang.org/ref/spec#Length_and_capacity) says len will never return < 0. Common sense says it shouldn't, but I thought it good to check for edge cases anyways.

diwakergupta commented on cmd/get.go:
```diff
@@ -27,11 +28,19 @@ import (
 
 func get(cmd *cobra.Command, args []string) (err error) {
+	if len(args) == 0 || len(args) > 2 {
+		return errors.New("`get` requires 1 or 2 arguments")
```
Maybe say "`get` requires `src` and/or `dst` arguments"?

waits replied:
That *could* be interpreted as meaning it's OK to pass `dst` without `src`. I don't think that's likely though.

diwakergupta commented on cmd/put.go:
```diff
@@ -56,10 +58,19 @@ func uploadChunked(r io.Reader, commitInfo *files.CommitInfo, sizeTotal int64) (
 
 func put(cmd *cobra.Command, args []string) (err error) {
+	if len(args) == 0 || len(args) > 2 {
+		return errors.New("`put` requires 1 or 2 arguments")
```
Same comment as above. Otherwise LG.

waits commented:
Added 8655d72 with your suggested error text.

diwakergupta commented:
Thanks!
